### ISSUE TITLE
Warn on non-reactive dependencies in computed

### ISSUE THREAD

**Issue by privatenumber** [vuejs/eslint-plugin-vue#1172]
**Please describe what the rule should do:**
Warn if a computed property references non-reactive VM properties, such as `$slots` or `$refs`.

**What category should the rule belong to?**

- [ ] Enforces code style
- [x] Warns about a potential error
- [ ] Suggests an alternate way of doing something
- [ ] Other (please specify:)

**Provide 2-3 code examples that this rule should warn about:**
I see this mistake frequently, but doesn't actually work because [slots aren't reactive](https://jsfiddle.net/hirokiosame/msyd41jn).

```vue
computed: {
  hasContentSlot() {
    return !!this.$slots.content;
  }
}
```

**Additional context**

**Comment by privatenumber** [comment]
Bumping this since it seems like it was missed

**Comment by ota-meshi** [comment]
Sorry, I missed it.
That rule sounds good to me :+1:

**Comment by renatodeleao** [comment]
+1!

I can't count how many times I had to mention this in code reviews and give [practical examples to showcase](https://codesandbox.io/s/non-reactive-slots-vue3-y35755?file=/src/App.vue:1189-1194): in conditional slot rendering context, there might be an illusion of reactivity when the computed value starts truthy.
